cluster-autoscaler/core/scaleup/resource/manager.go

```diff
@@ -180,6 +180,36 @@ func (m *Manager) ApplyLimits(ctx *context.AutoscalingContext, newCount int, res
 	return newCount, nil
 }
 
+// AllocateResources allocates the resources of required number of nodes from resources left and return true if resources are allocated successfully.
+func AllocateResources(left Limits, delta Delta, count int) bool {
+	allocate := func(dryRun bool) bool {
```

Comment:
My personal preference would be to make it a private function (and explicitly pass all input), rather than have it as func object.

Reply:
Updated, either looks good to me.